packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/index.js

```diff
@@ -74,27 +75,24 @@ function shimSmithySend(originalSend) {
     const smithySendArgs = getFunctionArguments(arguments);
     const command = smithySendArgs[0];
 
-    if (isActive) {
-      const serviceId = self.config && self.config.serviceId;
-      let awsProduct = serviceId && awsProducts.find(aws => aws.spanName === serviceId.toLowerCase());
-
-      if (awsProduct && awsProduct.supportsOperation(command.constructor.name)) {
-        return awsProduct.instrumentedSmithySend(self, originalSend, smithySendArgs);
-      } else {
-        // This code can be removed once all AWS SDK v3 instrumentations have been refactored to use the new approach
-        // introduced in https://github.com/instana/nodejs/pull/838 for kinesis. That is: Do not use an explicit
-        // operationsInfo/operationsMap map that restricts the traced operations to a subset of possible operations, but
-        // instead allow _all_ operations to be traced, using the operation name from `command.constructor.name` for
-        // span.data.$spanName.op. We plan to finish this refactoring before or with the next major release (3.x) of the
-        // @instana packages.
-        awsProduct = operationMap[smithySendArgs[0].constructor.name];
-        if (awsProduct) {
-          return awsProduct.instrumentedSmithySend(this, originalSend, smithySendArgs);
-        }
-
-        return originalSend.apply(this, smithySendArgs);
+    const serviceId = self.config && self.config.serviceId;
+    let awsProduct = serviceId && awsProducts.find(aws => aws.spanName === serviceId.toLowerCase());
+
+    if (awsProduct && awsProduct.supportsOperation(command.constructor.name)) {
+      return awsProduct.instrumentedSmithySend(self, isActive, originalSend, smithySendArgs);
+    } else {
+      // This code can be removed once all AWS SDK v3 instrumentations have been refactored to use the new approach
+      // introduced in https://github.com/instana/nodejs/pull/838 for kinesis. That is: Do not use an explicit
+      // operationsInfo/operationsMap map that restricts the traced operations to a subset of possible operations, but
+      // instead allow _all_ operations to be traced, using the operation name from `command.constructor.name` for
+      // span.data.$spanName.op. We plan to finish this refactoring before or with the next major release (3.x) of the
+      // @instana packages.
+      awsProduct = operationMap[smithySendArgs[0].constructor.name];
+      if (awsProduct) {
+        return awsProduct.instrumentedSmithySend(self, isActive, originalSend, smithySendArgs);
```

Comment:
_praise:_ I think this is actually better now, because it is more consistent. Each individual awsProduct now checks its `isActive` flag and either calls the original send method or the instrumented one. And we get rid of one `originalSend.apply(self, smithySendArgs);` here in index.js.

I like! :)